src/main/java/io/cryostat/net/reports/RecordingDescriptor.java

```diff
@@ -47,10 +47,13 @@
 class RecordingDescriptor {
     final ConnectionDescriptor connectionDescriptor;
     final String recordingName;
+    final String filter;
```

Comment:
`RecordingDescriptor` is meant to carry information about where the source recording file (which we're generating a report for) is located - the filter specification string doesn't seem like it fits well within this class with that context. Does that make sense? I think the `filter` field should be removed from here and passed around between method calls some other way.